lightning/src/ln/channelmanager.rs

```diff
@@ -2348,7 +2365,7 @@ impl<Signer: Sign, M: Deref, T: Deref, K: Deref, F: Deref, L: Deref> ChannelMana
 					ChannelUpdateStatus::DisabledStaged if chan.is_live() => chan.set_channel_update_status(ChannelUpdateStatus::Enabled),
 					ChannelUpdateStatus::EnabledStaged if !chan.is_live() => chan.set_channel_update_status(ChannelUpdateStatus::Disabled),
 					ChannelUpdateStatus::DisabledStaged if !chan.is_live() => {
-						if let Ok(update) = self.get_channel_update(&chan) {
+						if let Ok(update) = self.get_channel_update_for_broadcast(&chan) {
```

Comment:
Don't we also want to send a unicast in this case if the channel is private?

Reply:
Hmmm, I'd think it doesn't matter much - its not like our counterparty doesn't know that we're disconnected. I'm not sure what other nodes do, but we should correctly flag a channel as `!is_usable` in the `ChannelDetails` and thus not include it in invoices.